internal/config/config.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	"flag"
+	"os"
+	"strconv"
+)
+
+type ServerConfig struct {
+	Listen        string `yaml:"Listen" env-default:"localhost:8080"`
+	StorageDriver string `yaml:"StorageDriver" env-default:"mem"`
+}
+
+type AgentConfig struct {
+	Address        string `yaml:"address" env-default:"http://localhost:8080"`
+	PollInterval   int64  `yaml:"pollInterval" env-default:"2"`
+	ReportInterval int64  `yaml:"reportInterval" env-default:"10"`
+}
+
+func LoadServerConfig() ServerConfig {
+	var config ServerConfig
+	fromFlags := LoadServerConfigFromFlags()
+
+	address, ok := os.LookupEnv("ADDRESS")
```

Comment:
Я бы рекомнедовал добавлять префикс приложения к переменным окружения во избежание коллизий с другими переменными окружения.
```suggestion
	address, ok := os.LookupEnv("METRICS_ADDRESS")
```